addons/test_mail/tests/test_message_post.py

```diff
@@ -70,6 +70,28 @@ def setUp(self):
         # patch registry to simulate a ready environment; see ``_message_auto_subscribe_notify``
         self.patch(self.env.registry, 'ready', True)
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_manual_send_user_notification_email_from_queue(self):
+        """ Test sending a mail from the queue that is not related to the admin user sending it.
+        Should be sent without errors. """
+
+        mail = self.env['mail.mail'].create({
+            'email_from': self.user_employee.partner_id.email,
+            'email_to': self.partner_1.email,
+            'subject': 'This should be a subject',
+            'body_html': '<p>Test</p>',
+            'message_type': 'comment',
+            'state': 'outgoing',
+        })
+
+        self.assertNotIn(self.user_admin.partner_id, mail.partner_ids, "Our admin user should not be within the partner_ids")
+
+        with self.mock_mail_gateway():
+            mail.with_user(self.user_admin)._send()
+
+        self.assertEqual(mail.state, 'sent', 'Email should be in the sent state after manual send')
+        self.assertEqual(mail.write_uid, self.user_admin, 'The email should have been sent by the admin user')
```

Comment:
Globally you don't want to add tests in "common" test classes as they'll be run each time the class is inherited from. Would move to `TestMessagePost`.

Here's a one-line for the test that's a better reflection of the actual issue (as seen with calendar) as far as I can tell `env['res.partner'].browse(partner_demo_id).message_notify(subject="test", body="test", partner_ids=[partner_demo_id]).mail_ids.with_user(user_admin).send()`

i.e.
- use message_notify (creates a message with message_type="user_notification")
- try to send the linked email as a system user (you don't have access to the message due to custom rules)

Will ask exactly what we want to do, would be inclined to just sudo the write on message_id as it's anyway "too late" if we get to the point. We'll see :)